src/mailroom.py

```diff
@@ -0,0 +1,122 @@
+            for donor in sorted_donors:
+                print (donor)
+        elif choice == 'menu':
+            print ('\n \n')
+            main()
+        elif choice.lower() in donors:
+            chosen_donor = choice.lower()
+            input_prompt = choice + ' has already donated.'
+            input_prompt += '  Please enter a new donation:  '
+            donation = input(input_prompt)
+            new_donation = add_donation(donors, chosen_donor, donation)
+            print (print_email(donors, chosen_donor, new_donation))
+            main()
+        elif choice.lower() not in donors:
+            chosen_donor = choice.lower()
+            donors[chosen_donor] = []
+            input_prompt = chosen_donor + ' has been added.  Please'
+            input_prompt += ' submit a donation:  '
+            donation = input(input_prompt)
+            new_donation = add_donation(donors, chosen_donor, donation)
+            print (print_email(donors, chosen_donor, new_donation))
+            main()
+
+
+def create_sorted_list(donors):
+    """Create a sorted list of donors."""
+    sorted_list = []
+    for donor in sorted(map(lambda x: x.capitalize(), list(donors.keys()))):
+        donor = donor.split(' ')
+        sorted_list.append(' '.join(map(lambda x: x.capitalize(), donor)))
+    return sorted_list
+
+
+def add_donation(donors, donor, donation):
+    """Add donation for a speciic donor."""
+    invalid_donation = True
+    while invalid_donation:
+        try:
+            invalid_donation = float(donation)
+            donors[donor].append(float(donation))
+            invalid_donation = False
+        except ValueError:
+            input_prompt = 'Please enter a valid number: '
+            donation = input(input_prompt)
+    return donation
+
+
+def print_email(donors, donor, donation):
+    """Print email to console."""
+    donor_capital = ' '.join(map(lambda x: x.capitalize(), donor.split(' ')))
+    email = "Dear {0},\n \n \n"
+    email += 'Thanks so much for your donation of ${1}.  We love you!'
+    email += ' Please feel free to donate again.'
+    email += '\n \nLove, \n \n CodeFellows\n\n\n'
+    return email.format(donor_capital, donation)
+
+
+def create_a_report(donors):
+    """Create a donor report with list of donors, count, average and sum."""
+    output = 'Donor Name--------------Count-----Average--------Sum\n'
+    output += ('-' * 62) + '\n'
+    donor_ordered = [[k, sum(v)] for k, v in donors.items()]
+    donor_ordered.sort(key=lambda x: -x[1])
+    for donor in donor_ordered:
+        donations = donors[donor[0]]
+        donation_count = str(len(donations))
+        donation_sum = str(format(sum(donations), '.2f'))
+        donation_avg = str(format(float(sum(donations)) / len(donations), '.2f'))
+        name = donor[0].split(' ')
+        name = ' '.join(list(map(lambda x: x.capitalize(), name)))
+
+        output += '{0}{1}{2}{3}{4}{5}{6}{7}\n'.format(' ' * 3,
```

Comment:
Read the docs on `str.format` a little more closely. You can use the string formatting syntax to create automatically sized and justified columns!